### PR TITLE
Basic appointment management

### PR DESCRIPTION
**Pull request by qangdev**
- Calendar showing appointments
- Past appointments in gray, current and future ones in blue
- Modal to add new appointments
- Click on appointments on calendar to popup edit modal
- Delete button is in edit modal

### CODE REVIEW

File: app/models/shops.py [hunk 1]
```diff
@@ -45,3 +45,7 @@ def get_by_name(name):
         raise ErrNotFound(f"no data found for the provided name: {name}")
 
     return result[0]
+
+
+def all():
+    return shops.all()
```
Comment (mohanadkaleia) [inline app/models/shops.py:51]: maybe for the future we can take out changes that are not related to the ticket to a separate smaller pr :)

Comment (mohanadkaleia) [reply app/models/shops.py:51]: Thank you for adding this method :)

File: app/static/js/appointments.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+
+let appointment = {};
+appointment.data = [];
+
+$(document).ready(() => {
+  console.log(":>",appointment.data)
```
Comment (mohanadkaleia) [inline app/static/js/appointments.js]: is this for debugging?

File: app/static/js/appointments.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+          url: "/admin/appointments/view/upsert",
+          type: "POST",
+          data: {id: info.event.id}
+        });
+        $("#appointmentUpsertModalLabel").text("Edit Appointment");
+        $("#appointmentUpsertModal").find(".modal-body").html(formContent);
+        $("#appointmentUpsertDelForm").attr('action', `/admin/appointments/${info.event.id}/delete`)
+        $("#appointmentUpsertModal").modal('show');
+      })();
+      // change the border color just for fun
```
Comment (mohanadkaleia) [inline app/static/js/appointments.js]: :D

File: app/static/js/appointments.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      // info.el.style.borderColor = 'red';
+    },
+    dateClick: function() {
+      alert('a day has been clicked!');
```
Comment (mohanadkaleia) [inline app/static/js/appointments.js]: do we need this one?

File: app/views/__init__.py [hunk 5]
```diff
@@ -2,10 +2,12 @@
 from .auth import bp as auth_bp
 from .services_adm import bp as service_bp
 from .vehicles_adm import bp as vehicle_bp
+from .appointments_adm import bp as appointment_bg
```
Comment (mohanadkaleia) [inline app/views/__init__.py]: do you mean `appointment_bp`? just to make it consistent with other blueprints

File: app/views/appointments_adm.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    return render_template(
+        "admin/appointments/index.html",
+        active_nav="appointments",
+        appointments=json.dumps(events),
+    )
+
+
+@bp.route("/admin/appointments/view/upsert", methods=["POST"])
+def admin_render_upsert_form():
+    shops = shop_model.all()
```
Comment (mohanadkaleia) [inline app/views/appointments_adm.py:42]: curious, why we are pulling all the shops in here? is it possible to get the shop id from the form?

File: app/views/home.py [hunk 7]
```diff
@@ -41,7 +41,7 @@ def get_vehicle(id):
         # Search for the vehicle and render it
         data = vehicles_model.get(id)
         return render_template("vehicle_details.html", config=config, vehicle=data)
-    except vehicles_model.ErrVehicleNotFound:
+    except vehicles_model.ErrNotFound:
```
Comment (mohanadkaleia) [inline app/views/home.py:44]: nice catch!

**Comment by mohanadkaleia** [comment]
I tested the user interface little bit, here is a small feed back:
1. Is there any timezone set for the calendar plugin? it shows the current date is 22 of Sept but my time is still 21 of Sept :D 
2. Clicking on the empty days shows an alert window (maybe you created it for debugging only?) can you remove it. instead we can trigger the add new function

Other than that it looks great.. thank you!

File: app/views/home.py (hunk 7 above)
Comment (qangdev) [reply app/views/home.py:44]: No problem :D

File: app/views/appointments_adm.py (hunk 6 above)
Comment (qangdev) [reply app/views/appointments_adm.py:42]: That route is using to render the form.
I get all shops and services records to populate shop and service dropdown options in upsert_form template

File: app/views/__init__.py (hunk 5 above)
Comment (qangdev) [reply app/views/__init__.py]: Nice catch! Thanks. The p key is so closed to the g key. That's the reason why Ha ha

File: app/static/js/appointments.js (hunk 4 above)
Comment (qangdev) [reply app/static/js/appointments.js]: No, I'll remove it in the next commit

**Comment by mohanadkaleia** [comment]
I think it looks good! gonna merge it thank you!!!

**Pull request merged**